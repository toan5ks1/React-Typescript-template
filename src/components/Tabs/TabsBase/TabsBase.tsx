import { Tabs as MuiTabs, TabsProps as MuiTabsProps } from "@mui/material";

export interface TabsBaseProps extends MuiTabsProps {}

const TabsBase = ({ children, ...rest }: TabsBaseProps) => {
    return <MuiTabs {...rest}>{children}</MuiTabs>;
};

export default TabsBase;
