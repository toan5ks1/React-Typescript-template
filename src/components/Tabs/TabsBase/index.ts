export { default } from "./TabsBase";
export * from "./TabsBase";
