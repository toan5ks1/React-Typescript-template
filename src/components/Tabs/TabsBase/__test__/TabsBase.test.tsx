import { Tab } from "@mui/material";

import TabsBase, { TabsBaseProps } from "../TabsBase";

import { fireEvent, render, RenderResult, screen } from "@testing-library/react";

describe("<TabsBase />", () => {
    const tabTestId: string = "Tab";
    let wrapper: RenderResult;

    const defaultProps: TabsBaseProps = {
        indicatorColor: "primary",
        textColor: "primary",
        children: [
            <Tab label="Tab 1" key={0} data-testid="Tab" />,
            <Tab label="Tab 2" key={1} data-testid="Tab" />,
        ],
        value: 0,
        onChange: jest.fn(),
    };

    beforeEach(() => {
        wrapper = render(<TabsBase {...defaultProps} />);
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should render correctly UI", () => {
        const defaultTab = screen.queryAllByTestId(tabTestId)[0];
        const otherTab = screen.queryAllByTestId(tabTestId)[1];

        expect(defaultTab).toBeInTheDocument();
        expect(defaultTab).toHaveAttribute("aria-selected", "true");
        expect(defaultTab).toHaveTextContent("Tab 1");

        expect(otherTab).toBeInTheDocument();
        expect(otherTab).toHaveAttribute("aria-selected", "false");
        expect(otherTab).toHaveTextContent("Tab 2");
    });

    it("should clickable on tab", () => {
        const otherTab = screen.queryAllByTestId(tabTestId)[1];

        fireEvent.click(otherTab);
        expect(defaultProps.onChange).toBeCalled();
    });
});
