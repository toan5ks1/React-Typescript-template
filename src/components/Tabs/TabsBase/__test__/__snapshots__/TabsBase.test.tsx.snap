// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabsBase /> should match snapshot 1`] = `
.emotion-0 {
  overflow: hidden;
  min-height: 48px;
  -webkit-overflow-scrolling: touch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:599.95px) {
  .emotion-0 .MuiTabs-scrollButtons {
    display: none;
  }
}

.emotion-1 {
  position: relative;
  display: inline-block;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  white-space: nowrap;
  overflow-x: hidden;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.25;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  max-width: 360px;
  min-width: 90px;
  position: relative;
  min-height: 48px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 12px 16px;
  overflow: hidden;
  white-space: normal;
  text-align: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3.Mui-selected {
  color: #1976d2;
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-7 {
  position: absolute;
  height: 2px;
  bottom: 0;
  width: 100%;
  -webkit-transition: all 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: all 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: #1976d2;
}

<div>
  <div
    class="MuiTabs-root emotion-0"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed emotion-1"
      style="overflow: hidden; margin-bottom: 0px;"
    >
      <div
        class="MuiTabs-flexContainer emotion-2"
        role="tablist"
      >
        <button
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected emotion-3"
          data-testid="Tab"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab 1
          <span
            class="MuiTouchRipple-root emotion-4"
          />
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary emotion-3"
          data-testid="Tab"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab 2
          <span
            class="MuiTouchRipple-root emotion-4"
          />
        </button>
      </div>
      <span
        class="MuiTabs-indicator emotion-7"
        style="left: 0px; width: 0px;"
      />
    </div>
  </div>
</div>
`;
